pycbc/events/status.py

```diff
@@ -0,0 +1,252 @@
+import json, urllib
+
+def parse_veto_definer(veto_def_filename):
+    """ Parse a veto definer file from the filename and return a dictionary 
+    indexed by ifo and veto definer category level.
+    
+    Parameters
+    ----------
+    veto_def_filename: str
+        The path to the veto definer file
+    
+    Returns:
+        parsed_definition: dict   
+            Returns a dictionary first indexed by ifo, then category level, and
+            finally a list of veto definitions.
+    """
+    from glue.ligolw import ligolw, table, lsctables, utils as ligolw_utils
+    from glue.ligolw.ligolw import LIGOLWContentHandler as h
+    lsctables.use_in(h)
+    
+    indoc = ligolw_utils.load_filename(veto_def_filename, False, contenthandler=h)
+    veto_table = table.get_table(indoc, 'veto_definer')
+    
+    ifo = veto_table.getColumnByName('ifo')
+    name = veto_table.getColumnByName('name')
+    version = numpy.array(veto_table.getColumnByName('version'))
+    category = numpy.array(veto_table.getColumnByName('category'))
+    start = numpy.array(veto_table.getColumnByName('start_time'))
+    end = numpy.array(veto_table.getColumnByName('end_time'))
+    start_pad = numpy.array(veto_table.getColumnByName('start_pad'))
+    end_pad = numpy.array(veto_table.getColumnByName('end_pad'))
+    
+    data = {}
+    for i in range(len(veto_table)):
+        if ifo[i] not in data:
+            data[ifo[i]] = {}
+        
+        if category[i] not in data[ifo[i]]:
+            data[ifo[i]][category[i]] = []
+            
+        veto_info = {'name': name[i],
+                     'version': version[i],
+                     'start': start[i],
+                     'end': end[i],
+                     'start_pad': start_pad[i],
+                     'end_pad': end_pad[i],
+                     }
+        data[ifo[i]][category[i]].append(veto_info)
+    return data
+
+def query_flag(ifo, segment_name, start_time, end_time,
+               source='any', server="segments.ligo.org",
+               veto_definer=None):
+    """Return the times where the flag is active 
+
+    Parameters
+    ----------
+    ifo: string
+        The interferometer to query (H1, L1).
+    segment_name: string
+        The status flag to query from LOSC.
+    start_time: int 
+        The starting gps time to begin querying from LOSC
+    end_time: int 
+        The end gps time of the query
+    source: str, Optional
+        Choice between "GWOSC" or "dqsegdb". If dqsegdb, the server option may
+        also be given. The default is to try GWOSC first then try dqsegdb.
+    server: str, Optional
+        The server path. Only used with dqsegdb atm.
+    veto_definer: str, Optional
+        The path to a veto definer to define groups of flags which
+        themselves define a set of segments.
+
+    Returns
+    ---------
+    segments: glue.segments.segmentlist
+        List of segments
+    """
+    if source == 'GWOSC' or source == 'any':
```

Comment:
Source=`any` is not defined, but is the default?? What is this?

Reply:
Indeed it seems to mean `GWOSC`, which seems wrong.

Reply:
If you give 'any' it tries GWOSC first to find the flag, then proceeds to dqsegdb if that fails.